### PR TITLE
Automatically select another existing layout after deleting multiple layouts at once

### LINKED ISSUE #4454: "Your active layout was deleted" when multi-selecting to delete layouts
**Description**

- Version:
- OS:
- Data source (e.g. bag file, mcap file, rosbridge, ROS 1/2 native):
- ROS distro (if applicable):

**Steps To Reproduce**
- Cmd-click to select multiple layouts
- Click `...` > Delete and delete the layouts

**Current Experience**
The "Your active layout deleted" toast *always* appears when deleting multiple layouts.
This differs from the behavior when deleting a single selected layout.
image

**Desired Experience**
The behavior should be consistent with deleting a single layout (when not using multi-selection).
Automatically select another layout that was not deleted, or maybe create a new default layout?

### PR DESCRIPTION
**Pull request by osaro-imohe**
**User-facing changes**
Not worth calling out in release notes

**Description**
This makes multi-layout delete consistent with single layout delete.

Resolves #4454

### CODE REVIEW

**Comment by osaro-imohe** [comment]
@jtbandes

**Comment by jtbandes** [comment]
We need to discuss this in a product review. @foxymiles is not a fan of randomly selecting an existing layout; we talked about some other ideas like showing a getting started screen that lets you easily create a new layout or choose an existing one, without having the extra click to open the sidebar.

**Review by amacneil — APPROVED** [review]
👍 consistency

**Comment by amacneil** [comment]
Deleting a single active layout seems to select the first layout in the list (at least according to my experiments - I can't find the code)
Deleting multiple layouts including the active layout selects the last layout in the list

We should make these consistent - always select the first layout in the list

**Comment by amacneil** [comment]
@jtbandes for code review and signoff

**Comment by osaro-imohe** [comment]
Good point
I'll get that

File: packages/studio-base/src/providers/CurrentLayoutProvider/index.tsx [hunk 1]
```diff
@@ -242,8 +242,9 @@ export default function CurrentLayoutProvider({
       }
 
       if (event.layoutId === layoutStateRef.current.selectedLayout.id) {
-        await setSelectedLayoutId(undefined);
-        enqueueSnackbar("Your active layout was deleted.", { variant: "warning" });
+        const layouts = await layoutManager.getLayouts();
+        const currLayout = layouts[0];
+        await setSelectedLayoutId(currLayout?.id ?? undefined);
```
Comment (jtbandes) [inline packages/studio-base/src/providers/CurrentLayoutProvider/index.tsx:247]: The `?? undefined` is redundant.
```suggestion
        await setSelectedLayoutId(layouts[0]?.id);
```

**Review by jtbandes — APPROVED** [review]

**Comment by jtbandes** [comment]
Thanks for the contribution!

**Pull request merged**